### PR TITLE
win, mmap uses VirtualAlloc instead of anon mem mapped file

### PR DESCRIPTION
**Pull request by BBasile**
fixes a mess with windows handle that had to be static and not static at the same time

### CODE REVIEW

File: std/experimental/allocator/mmap_allocator.d [hunk 1]
```diff
@@ -50,47 +50,24 @@ struct MmapAllocator
...
-                hiSz = HIWORD(bytes);
-            }
-            else loSz = bytes;
-
-            auto fh = CreateFileMappingA(INVALID_HANDLE_VALUE,
-                LPSECURITY_ATTRIBUTES.init, PAGE_READWRITE, hiSz, loSz, null);
-            if (fh == INVALID_HANDLE_VALUE) return null;
-
-            auto p = MapViewOfFile(fh, FILE_MAP_ALL_ACCESS, 0u, 0u, bytes);
+            auto p = VirtualAlloc(null, bytes, MEM_COMMIT, PAGE_EXECUTE_READWRITE);
```
Comment (DmitryOlshansky) [inline std/experimental/allocator/mmap_allocator.d]: EXECUTE might be a security concern if not requested explicitly. Go for READWRITE, leave EXECUTE for JIT compiler internals.

Comment (BBasile) [inline std/experimental/allocator/mmap_allocator.d]: Right, thks, this was a clearly mistake.

**Comment by DmitryOlshansky** [comment]
LGTM

File: std/experimental/allocator/mmap_allocator.d [hunk 2]
```diff
@@ -50,47 +50,24 @@ struct MmapAllocator
...
         bool deallocate(void[] b) shared
         {
-            int result = true;
+            bool result = true;
             if (b.ptr)
-                result &= UnmapViewOfFile(b.ptr);
-            if (f != INVALID_HANDLE_VALUE)
-                result &= CloseHandle(cast(HANDLE) f);
-            return result != 0;
+                result &= VirtualFree(b.ptr, 0, MEM_RELEASE) != 0;
```
Comment (andralex) [inline std/experimental/allocator/mmap_allocator.d]: no need for &=, assignment suffices. Better yet: `return b.ptr is null || VirtualFree(b.ptr, 0, MEM_RELEASE) != 0;`

**Comment by BBasile** [comment]
note that I wont update the DDOC comment. However it's clearly outdated. It's a matter of style. Either the common traits has to be found for both versions or something like ___under Linux mmap_allocator does this...under Windows it does that...___

**Comment by andralex** [comment]
oh I thought I'd pulled that already...

**Pull request merged**